CRM/Contribute/BAO/Contribution.php

```diff
@@ -183,9 +187,10 @@ public static function add(&$params, $ids = []) {
     if ($contributionID && $setPrevContribution) {
       $params['prevContribution'] = self::getOriginalContribution($contributionID);
     }
+    $previousContributionStatus = ($contributionID && !empty($params['prevContribution'])) ? CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', (int) $params['prevContribution']->contribution_status_id) : NULL;
```

Comment:
I don't really follow if there is a functional change here or not. From a code readability perspective, I like the creation of the $previousContributionStatus variable even if it is only used once. But it's a bit more complicated how you are doing it. Is the reason for only checking the previous contribution status if there is a contribution and a previous contribution mainly to improve performance? Or is there a possible error that's being avoided? Sorry, I'm trying to help with a review but I'm struggling a bit.

Reply:
It's primarily about making the IF lower down readable in this case.